### PR TITLE
Format our source

### PR DESCRIPTION
**Pull request by 333fred**
This adds the roslyn .editorconfig for the repo. I then ran the formatter over all documents in the solution (thanks @TyOverby for https://github.com/TyOverby/roslyn-diff-formatter, which I hacked up). Finally, I added the .editorconfig and README to the solution for ease of editing. Tagging @dotnet/roslyn-analysis for review.

### CODE REVIEW

File: RoslynAnalyzers.sln [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26228.0
+VisualStudioVersion = 15.0.26424.2
```
Comment (heejaechang) [inline RoslynAnalyzers.sln]: do we care about the version?

**Comment by heejaechang** [comment]
👍

File: RoslynAnalyzers.sln (hunk 1 above)
Comment (333fred) [inline RoslynAnalyzers.sln]: Not really. VS will just update it to whatever version you're running when you open the solution anyway.

**Comment by jinujoseph** [comment]
great 👍

**Pull request merged**